apis/google_books.py

```diff
@@ -0,0 +1,135 @@
+    resp = requests.get(ol_url)
+    data = resp.json()
+    if "bio" in data:
+        author_data["about"] = data["bio"]
+    if "birth_date" in data:
+        author_data["born_at"] = data["birth_date"]
+    if "death_date" in data:
+        author_data["died_at"] = data["death_date"]
+    author_data["large_image_url"] = f"https://covers.openlibrary.org/a/olid/{olid}-L.jpg"
+    author_data["name"] = data["name"]
+
+    return author_data
+
+
+def get_google_isbn(google_id: str) -> str:    
+    isbn_data = {}
+    g_url = f'https://www.googleapis.com/books/v1/volumes/{google_id}?key={google_key}'
+    resp = requests.get(g_url)    
+    data = resp.json()["volumeInfo"]
+    
+    isbn_data["isbn10"] = data["industryIdentifiers"][0]["identifier"]
+    isbn_data["isbn13"] = data["industryIdentifiers"][1]["identifier"]
+
+    return isbn_data
+
+
+def fetch_amazon(url: str) -> Dict:    
+    isbn = url.split("/")[-1]
+    volume_url = search_book(isbn)
+    api_data = fetch_google(volume_url)    
+
+    return api_data
+
+
+def fetch_google(url: str) -> Dict:
+    # get isbn from amazon url    
+    api_data = {}
+    gid = url.split("/")[-1]    
+    g_url = f'https://www.googleapis.com/books/v1/volumes/{gid}?key={google_key}'
+    resp = requests.get(g_url)
+    metadata = resp.json()    
+    data = metadata["volumeInfo"]
+
+    if "description" in data:
+        api_data["description"] = data["description"]
+    api_data["canonicalVolumeLink"] = data["canonicalVolumeLink"]
+    api_data["google_id"] = gid
+    api_data["google_url"] = data["infoLink"]
+    api_data["form"] = "text"
+    api_data["image_url"] = data["imageLinks"]["extraLarge"] if "extraLarge" in data[
+        "imageLinks"] else data["imageLinks"]["large"] if "large" in data[
+            "imageLinks"] else data["imageLinks"]["medium"] if "medium" in data[
+                "imageLinks"] else data["imageLinks"]["small"] if "small" in data[
+                    "imageLinks"] else "https://ik.imagekit.io/analogue/content_placeholder_1_85wLgUnbx.jpg"
+    api_data["isbn10"] = data["industryIdentifiers"][0]["identifier"]
+    api_data["isbn13"] = data["industryIdentifiers"][1]["identifier"]
+    api_data["language"] = data["language"]
+    #api_data["medium"] = data["printType"]
+    api_data["medium"] = "book"
+    api_data["publisher"] = data["publisher"]
+    api_data[
+        "publication_date"
+    ] = data["publishedDate"].replace('-', '/')
+    api_data["title"] = data["title"]
+    api_data["origin"] = "amazon.com"
+    api_data["origin_url"] = f"https://www.amazon.com/dp/{api_data['isbn10']}"
+
+    ol_url = f'http://openlibrary.org/api/volumes/brief/isbn/{api_data["isbn10"]}.json'    
+    ol_resp = requests.get(ol_url).json()
+    if ol_resp != []:
+        ol_metadata = ol_resp["records"]
+        ol_key = list(ol_metadata.keys())[0]
+        ol_data = ol_metadata[ol_key]
+
+        api_data["ol_work_id"] = ol_data["details"]["details"]["works"][0]["key"].split(
```

Comment:
@joelrojo currently not parsing this openlibrary data back in mainframe, but kept it in mercury in case we wanted to use it for keeping track of editions later